core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3955,11 +3960,473 @@ class KafkaApis(val requestChannel: RequestChannel,
+
+      val fetchMaxBytes = Math.min(Math.min(shareFetchRequest.maxBytes, config.fetchMaxBytes), maxQuotaWindowBytes)
+      val fetchMinBytes = Math.min(shareFetchRequest.minBytes, fetchMaxBytes)
+
+      val clientMetadata: Optional[ClientMetadata] =
+        Optional.of(new DefaultClientMetadata(
+          CommonClientConfigs.DEFAULT_CLIENT_RACK,
+          clientId,
+          request.context.clientAddress,
+          request.context.principal,
+          request.context.listenerName.value))
+
+      // TODO : Change the dummy values to actual values after the required implementations are completed
+      val params = new FetchParams(
+        versionId,
+        FetchRequest.FUTURE_LOCAL_REPLICA_ID,
+        -1,
+        shareFetchRequest.maxWait,
+        fetchMinBytes,
+        fetchMaxBytes,
+        FetchIsolation.HIGH_WATERMARK,
+        clientMetadata
+      )
+
+      val interestingTopicPartitions: util.List[TopicIdPartition] = new util.ArrayList[TopicIdPartition]()
+      val iterator = interestingWithMaxBytes.entrySet().iterator()
+      while (iterator.hasNext) {
+        val entry = iterator.next()
+        interestingTopicPartitions.add(entry.getKey)
+      }
+
+      // call the share partition manager to fetch messages from the local replica.
+      try{
+        val responsePartitionData = sharePartitionManager.fetchMessages(
+          groupId,
+          shareFetchRequest.data.memberId,
+          params,
+          interestingTopicPartitions,
+          interestingWithMaxBytes
+        ).get()
+        processResponseCallback(responsePartitionData.asScala.toMap)
+      } catch {
+        case e: Throwable => throw e
+      }
+    }
+  }
+
+  def handleAcknowledgements (
+                               request: RequestChannel.Request,
+                               topicNames : util.Map[Uuid, String],
+                               sharePartitionManager : SharePartitionManager,
+                               authorizedTopics: Set[String],
+                               groupId : String,
+                               memberId: String,
+                               areAcknowledgementsPiggyBacked : Boolean,
+                             ) : mutable.Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData] = {
+
+    val requestData : AbstractRequest = if (areAcknowledgementsPiggyBacked) {
+      request.body[ShareFetchRequest]
+    } else {
+      request.body[ShareAcknowledgeRequest]
+    }
+    val erroneous = mutable.Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData]()
+
+    var acknowledgementDataFromRequest : mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]] = null
+    if(areAcknowledgementsPiggyBacked) {
+      acknowledgementDataFromRequest = getAcknowledgeBatchesFromShareFetchRequest(requestData.asInstanceOf[ShareFetchRequest], topicNames, erroneous)
+    } else {
+      // TODO: handle acknowledgements from shareAcknowledge request
+    }
```

Comment:
Hmmm, this is not clean.

Why can't we have `handleAcknowledgements` API to receive `mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]]`? Then `ShareFetch` and `ShareAcknowledgeRequest` can send respective data to `handleAcknowledgements` method and we need not to use `asInstanceOf` or any type casting.